src/app/frontend/logs/logs_service.js

```diff
@@ -0,0 +1,57 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Service function for logs panel management.
+ * @return {{registerListener: Function, notifyPodChange: Function, notifyTextColorChange:
+ * Function}}
+ */
+export default function logsService() {
```

Comment:
I think if you preserve all parameters required to display logs in the url (e.g., /logs/:namespace/:replicaSet/:pod/:container) you dont need this service at all. This would save a lot of code and complexity.